classification/config.py

```diff
@@ -3,116 +3,188 @@
     Methods:
         get_config: returns an ArgumentParser with the default arguments
 """
-import sys
 import argparse
+import os
+import shutil
+import sys
+
 import git
+import yaml
 
-# Machine learning has a lot of arugments
-# pylint: disable=too-many-statements
-def get_config():
-    """ Returns a config variable with the command line arguments or defaults
+
+class Config():
+    """
+    Saves config from config files
+    Allows for customising runs
     """
-    parser = argparse.ArgumentParser()
+    def __init__(self):
+        """Constructor that runs after creating the singleton class
+        Post processing after reading config file
+        """
+        self.required_checks( "dataframe_csv")
+        self.required_checks( "data_path")
+        self.get_git_hash()
+        self.cli_values()
 
-    # Dataset settings
-    parser.add_argument('-df', '--dataframe', default='CHANGEME.csv', type=str)
-    parser.add_argument('-dp', '--data_path', default='./all_10_species', type=str)
+    def __new__(cls):
+        """
+        Constructor for Config
 
+        Takes data from config files and generates a singleton class
+        that can store all system vars
 
-    parser.add_argument('-st', '--offset_col', default='OFFSET', type=str)
-    parser.add_argument('-et', '--duration_col', default='DURATION', type=str)
-    parser.add_argument('-fn', '--file_name_col', default='FILE NAME', type=str)
-    parser.add_argument('-mi', '--manual_id_col', default='SCIENTIFIC', type=str)
+        returns a refrence to the singleton class
 
-    # Env settings
-    parser.add_argument('-tbs', '--train_batch_size', default=4, type=int)
-    parser.add_argument('-vbs', '--valid_batch_size', default=4, type=int)
-    parser.add_argument('-wbs', '--wandb_session', default="acoustic-species-reu2023", 
-                        type=str, help="wandb project name")
+        Intended to read config files
+        """
+        
+        # Set up singleton class design template
+        if not hasattr(cls, 'instance'):
+            cls.instance = super(Config, cls).__new__(cls)
+        else:
+            return cls.instance
 
-    parser.add_argument('-vdr', '--valid_dataset_ratio', default=0.4, type=float,
-                        help="ratio of validation dataset to run during epochs") 
+        #Set defaults config
+        with open('config.yml', 'r', encoding='utf-8') as file:
```

Comment:
think `config.yml` needs to be in the classification directory then - make sure documentation and README has this info

Reply:
Okie gonna rename them to config.yml and documentation/default_config.yml